### PR TITLE
update ASan content

### PR DESCRIPTION
**Pull request by randyh62**
cherry pick PR #3284

### CODE REVIEW

**Review by Rmalavally — APPROVED** [review]
Looks good, @randyh62. Added a couple of minor comments.

**Review by yhuiYH — CHANGES_REQUESTED** [review]
Do we really need to apply this to 6.0.0?

File: docs/conceptual/using-gpu-sanitizer.md [hunk 1]
```diff
@@ -86,16 +96,25 @@ If it does not appear, when executed the application will quickly output an ASan
...
+
+* `quarantine_size_mb=N default 256`
+
+  This option defines the number of megabytes (MB) `N` of memory that the ASan runtime will hold after it is `freed` to detect use-after-free situations. This memory is unavailable for other purposes. The default of 256 MB may be too small to detect some use-after-free situations, especially given that the large size of many GPU memory allocations may push `freed` allocations out of quarantine before the attempted use.
+
+  :::{note}
+  Setting the value of `quarantine_size_mb` larger may enable more problematic uses to be detected, but at the cost of reducing memory available for other purposes.
+  :::
 
 ## Runtime overhead
```
[the remark this answers is not in the document]
Comment (randyh62) [reply docs/conceptual/using-gpu-sanitizer.md:119]: No, there is content here, just not being displayed.

**Comment by randyh62** [comment]
Yes, the content is needed in this release.

**Comment by peterjunpark** [comment]
Wording suggestions (etc.) have to be finalized in the PR for develop... before this stage

File: docs/conceptual/using-gpu-sanitizer.md [hunk 2]
```diff
@@ -198,7 +217,7 @@ This is solved by setting environment variable `LD_PRELOAD` to the path to the A
 amdclang++ -print-file-name=libclang_rt.asan-x86_64.so
 ```
 
-It is also recommended to set the environment variable `HIP_ENABLE_DEFERRED_LOADING=0` before debugging HIP applications.
+You should also set the environment variable `HIP_ENABLE_DEFERRED_LOADING=0` before debugging HIP applications.
```
[the remark this answers is not in the document]
Comment (peterjunpark) [reply docs/conceptual/using-gpu-sanitizer.md:220]: Wording suggestions (etc.) should be finalized in the PR for develop... before this stage

**Pull request merged**